tls/conn.go

```diff
@@ -626,7 +628,13 @@ Again:
 			c.in.freeBlock(b)
 			goto Again
 		case alertLevelError:
-			c.in.setErrorLocked(&net.OpError{Op: "remote error", Err: alert(data[1])})
+			var e error
+			if alert(data[1]) == alertHandshakeFailure {
```

Comment:
This seemed simpler and more like a common Go pattern than some sort of deep comparison, but see the code depending on it in cfssl/scan/tls_handshake.go.